crates/optimism/chainspec/src/chain_metadata.rs

```diff
@@ -0,0 +1,312 @@
+/// Referring here as `ChainMetadata` to avoid confusion with `ChainConfig`.
+/// Find configs here: `<https://github.com/ethereum-optimism/superchain-registry/tree/main/superchain/configs>`
+/// This struct is stripped down to only include the necessary fields. We use JSON instead of
+/// TOML to make it easier to work in a no-std environment.
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub(crate) struct ChainMetadata {
+    pub chain_id: ChainId,
+    pub hardforks: HardforkConfig,
+    pub optimism: Option<OptimismConfig>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub(crate) struct HardforkConfig {
+    pub canyon_time: Option<u64>,
+    pub delta_time: Option<u64>,
+    pub ecotone_time: Option<u64>,
+    pub fjord_time: Option<u64>,
+    pub granite_time: Option<u64>,
+    pub holocene_time: Option<u64>,
+    pub isthmus_time: Option<u64>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub(crate) struct OptimismConfig {
+    pub eip1559_elasticity: u64,
+    pub eip1559_denominator: u64,
+    pub eip1559_denominator_canyon: Option<u64>,
+}
+
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct ChainConfigExtraFields {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub bedrock_block: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub regolith_time: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub canyon_time: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub delta_time: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ecotone_time: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fjord_time: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub granite_time: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub holocene_time: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub isthmus_time: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub optimism: Option<ChainConfigExtraFieldsOptimism>,
+}
+
+// Helper struct to serialize field for extra fields in ChainConfig
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct ChainConfigExtraFieldsOptimism {
+    pub eip1559_elasticity: u64,
+    pub eip1559_denominator: u64,
+    pub eip1559_denominator_canyon: Option<u64>,
+}
+
+impl From<&OptimismConfig> for ChainConfigExtraFieldsOptimism {
+    fn from(value: &OptimismConfig) -> Self {
+        Self {
+            eip1559_elasticity: value.eip1559_elasticity,
+            eip1559_denominator: value.eip1559_denominator,
+            eip1559_denominator_canyon: value.eip1559_denominator_canyon,
+        }
+    }
+}
+
+/// Returns a [`ChainConfig`] filled from [`ChainMetadata`] with extra fields and handling
```

Comment:
same here, we probably want to use this type right? https://github.com/op-rs/kona/blob/a1899da1cfc0932b2f42fb799abdc5b558c1ba39/crates/protocol/genesis/src/chain/config.rs#L13-L106